File: backend/app/schemas/collection.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class CollectionCreate(BaseModel):
    name: str
    description: str | None = None


class CollectionUpdate(BaseModel):
    name: str | None = None
    description: str | None = None


class CollectionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    description: str | None
    owner_id: uuid.UUID
    qdrant_collection_name: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/collection.ts
export interface Collection {
  id: string
  name: string
  description: string | null
  owner_id: string
  qdrant_collection_name: string
  created_at: string
  updated_at: string
}

export interface CollectionCreate {
  name: string
  description?: string
}

export interface CollectionUpdate {
  name?: string
  description?: string
}
